Signaling/app/src/main/java/com/tokbox/android/tutorials/signaling/ChatMessage.java

```diff
@@ -0,0 +1,33 @@
+package com.tokbox.android.tutorials.signaling;
+
+public class ChatMessage {
+
+    private String mMessageText;
+    private Boolean mRemote;
+
+    public ChatMessage(String messageText) {
+        mMessageText = messageText;
+        mRemote = false;
+    }
+
+    public static ChatMessage fromData(String messageData) {
+        return new ChatMessage(messageData);
+    }
+
+    public String getMessageText() {
+        return mMessageText;
+    }
+
+    public Boolean getRemote() {
+        return mRemote;
+    }
+
+    public void setRemote(Boolean remote) {
+        mRemote = remote;
+    }
+
+    @Override
+    public String toString() {
```

Comment:
@hbaqai regarding the ArrayAdapter, the ViewHolder pattern could be a good practice for it to avoid to call to findViewById() several times.